### DISCUSSION TITLE
Plot a shape

### DISCUSSION THREAD

**Discussion by kinnetekLluis** [oxyplot/oxyplot#2010] [Q&A] [answered]
Hi, I am new to Oxyplot and I don't find info to do a simple shape filled with a color.

In my case I have the line defining the enclosure and I can fullfill the color by a lineseries going up and down, but I feel like it's not the way to go.

Here I present the line data I have:

<img width="1193" alt="raw" src="https://github.com/oxyplot/oxyplot/assets/113010916/749146ab-8f45-497e-af9a-5452a7113d46">

And what I wanted to do (I generate this with lines, which makes it ineficient if I use too many points):

<img width="547" alt="obj" src="https://github.com/oxyplot/oxyplot/assets/113010916/bfc0893c-5963-477f-b6af-42f89ac953ad">

Edit: I forgot to add the code! (list is the variable with the datapoints)

```
            var tmp = new PlotModel();
            var s1 = new LineSeries
            {
                StrokeThickness = 1,
                MarkerSize = 0,
                MarkerStroke = OxyColors.IndianRed,
                MarkerType = MarkerType.Plus,
                Color = OxyColor.FromArgb(155, 205, 92, 92),                
            };

            for (int i = 0; i < list[0].Count; i++)
            {
                s1.Points.Add(new DataPoint(list[0][i], list[1][i]));
            }

            tmp.Series.Add(s1);
            tmp.PlotType = PlotType.Cartesian;
            plotoxy.Model = tmp;
```

**Comment by kinnetekLluis** [comment, marked as answer]
I found a way pretty simple but I dont know if it's a good practice, using "PolygonAnnotation"...

```
var tmp = new PlotModel();
var s1 = new LineSeries
            {
                StrokeThickness = 1,
                MarkerSize = 0,
                MarkerStroke = OxyColors.IndianRed,
                MarkerType = MarkerType.Plus,
                Color = OxyColor.FromArgb(155, 205, 92, 92),                
            };

var Event = new PolygonAnnotation();

Event.Layer = AnnotationLayer.BelowAxes;
Event.StrokeThickness = 0;
Event.LineStyle = LineStyle.Automatic;

for (int i = 0; i < list[0].Count; i++)
            {
                s1.Points.Add(new DataPoint(list[0][i], list[1][i]));
                Event.Points.Add(new DataPoint(list[0][i], list[1][i]));
            }

tmp.Series.Add(s1);
tmp.PlotType = PlotType.Cartesian;
tmp.Annotations.Add(Event);
plotoxy.Model = tmp;
```

But I also dont know how to change the color in this case :S

**Reply by VisualMelon** [reply]
Nothing wrong with using a `PolygonAnnotation`. Use the `Fill` and `Stroke` properties to change the color of a `PolygonAnnotation`.

If you want it a series, you could use `AreaSeries`; I also started work on a `PolygonSeries`: https://github.com/oxyplot/oxyplot/issues/1751

What is your use-case? Always keen to know what people are trying to do that library doesn't support well so that we can focus our efforts on useful features.

**Reply by kinnetekLluis** [reply]
Oooh thanks! Now I got what I wanted and it's much lighter! The only thing I see using PolygonAnnotation is that I still need to provide the series if I want to keep the axis in the range of the values (if not, my annotation is not centered). I think that removing the series would improve the performance a bit more, but then I should manually adjust the axis and for now I don't know yet how to do it x)

I am studying the work envelop of a robot, which is the representation of how the robot can move (in this case) in 2 directions. I had the line because it's the limit were it can go, then I wanted to fill the enclosed space as to represent that the colored space the robot can move around without constrain. Usually, the workspace (or work envelop) is a close shape so for me, it would be really nice to have a direct tool to just fullfil the closed space!

In the future I will have to do this same plot but adding a heatmap representing the velocities on each point of the workeable space, I don't know if there is such tool here but it would be awesome! 

Thanks for your previous answer and in advance!

PD: one of the results!

<img width="527" alt="test1" src="https://github.com/oxyplot/oxyplot/assets/113010916/7591a52e-c8cf-4ebf-b8d7-3497c89427c5">

**Reply by VisualMelon** [reply]
You can specify the axis bounds manually by setting the `Minimum` and `Maximum` values, which may make more sense if e.g. you are talking about a well-defined space and want to judge what parts of it the envelope covers.

`AreaSeries` would otherwise provide comparable functionality if you specify `Points` but not `Points2`.

Regarding velocities, heatmaps are all square the moment, and there is no built in vector series. Again, PolygonAnnotation may be useful in this case.

Thanks for explicating the use case!

**Reply by kinnetekLluis** [reply]
Thanks, your contribution was on point! 

Keep the good work! And if you implement that, let me know :D

**Reply by VisualMelon** [reply]
You can copy the implementation from https://github.com/oxyplot/oxyplot/commit/36e923455bedc17e62a5e6ad1e83d875e0e9ae3d if it's of any use.
